### PR TITLE
Native vector types

### PR DESCRIPTION
**Pull request by tbennun**
- [x] Add dace.vector as a typeclass 
- [x] Add dace.vector to type inference
- [x] Use vector types in code generator
- [x] Use vector types in Vectorization xform
- [x] Redo vector length propagation in Vectorization xform
- [x] Remove veclen from memlets and memlet propagation

### CODE REVIEW

File: dace/memlet.py [hunk 1]
```diff
@@ -372,6 +364,10 @@ def try_initialize(self, sdfg: 'dace.sdfg.SDFG',
                 is_data_src = False
         self._is_data_src = is_data_src
 
+        # If subset is None, fill in with entire array
+        if (self.data is not None and self.subset is None):
+            self.subset = subsets.Range.from_array(sdfg.arrays[self.data])
```
Comment (definelicht) [inline dace/memlet.py:369]: What happens for Streams? Does it insert "0"?

File: tests/intel_fpga_test.sh [hunk 2]
```diff
@@ -69,15 +69,15 @@ run_all() {
     run_sample pipeline_scope pipeline_scope "\n" 
 
     # Test shift register abstraction with stencil code
-    run_sample fpga_stencil fpga_stencil_test "\n"
+    # run_sample fpga_stencil fpga_stencil_test "\n"
```
Comment (definelicht) [inline tests/intel_fpga_test.sh:72]: Oh shit

File: dace/memlet.py (hunk 1 above)
Comment (tbennun) [reply dace/memlet.py:369]: yes :) this relates to your issue from before.

File: dace/codegen/targets/cpu.py [hunk 3]
```diff
@@ -880,23 +885,21 @@ def memlet_definition(self,
                           local_name: str,
                           conntype: Union[data.Data, dtypes.typeclass] = None,
                           allow_shadowing=False):
+        assert conntype is not None
```
[the remark this answers is not in the document]
Comment (tbennun) [reply dace/codegen/targets/cpu.py:888]: Done.

**Pull request merged**